src/lib/autoinstall/autoinst_issues/issue.rb

```diff
@@ -0,0 +1,52 @@
+module Y2Autoinstallation
+  module AutoinstIssues
+    # Base class for autoinstallation problems while importing the
+    # AutoYaST configuration file.
+    #
+    # Y2Autoinstallation::AutoinstIssues offers an API to register
+    # and report related AutoYaST problems.
+    class Issue
+      include Yast::I18n
+
+      # @return [#parent,#section_name] Section where it was detected
+      attr_reader :section
+
+      # Return problem severity
+      #
+      # * :fatal: abort the installation.
+      # * :warn:  display a warning.
+      #
+      # @return [Symbol] Issue severity (:warn, :fatal)
+      # @raise NotImplementedError
+      def severity
```

Comment:
np: I've seen that you are redefining this method in the derived classes to something like:

```ruby
def severity
  @severity
end
```
Perhaps you would like to make that the default definition (and avoid redefining it). What about something like this here (and removing the redefined ones):

```ruby
def severity
  @severity || :warning
end
```